### ISSUE TITLE
eval_image_classifier -- error w/ inception_v4

### ISSUE THREAD

**Issue by scotshinderman** [tensorflow/models#685]
slim/inception_v4 (also v3)

both inception_v4 and v3 give an error while using eval_image_classifier:

models:

commit 4bcbe8e140a6c83c151b93567f33e60e486922bd

tensor flow:

commit c80ef0cd19a3ac9ad0657f617b4e7328f6a48ac3

(bazel build -c opt //tensorflow/tools/pip_package:build_pip_package)

`python eval_image_classifier.py  --alsologtostderr  --checkpoint_path=/Users/scot/Desktop/tf/data/inception_v4.ckpt      --dataset_dir=/Users/scot/Desktop/tf/data/cifar10      --dataset_name=cifar10     --dataset_split_name=test     --model_name=inception_v4`

INFO:tensorflow:Scale of 0 disables regularizer.
WARNING:tensorflow:From eval_image_classifier.py:157 in main.: streaming_recall_at_k (from tensorflow.contrib.metrics.python.ops.metric_ops) is deprecated and will be removed after 2016-11-08.

Instructions for updating:

Please use streaming_sparse_recall_at_k, and reshape labels from [batch_size] to [batch_size, 1].
INFO:tensorflow:Evaluating /Users/scot/Desktop/tf/data/inception_v4.ckpt
INFO:tensorflow:Starting evaluation at 2016-11-24-03:10:25

**INFO:tensorflow:Error reported to Coordinator: <class 'tensorflow.python.framework.errors_impl.InvalidArgumentError'>, Assign requires shapes of both tensors to match. lhs shape= [1536,10] rhs shape= [1536,1001]**

[[Node: save/Assign_23 = Assign[T=DT_FLOAT, _class=["loc:@InceptionV4/Logits/Logits/weights"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/cpu:0"](InceptionV4/Logits/Logits/weights, save/RestoreV2_23)]]

let me know if you need more info -- thanks!

**Comment by SirNeural** [comment]
From what it looks like, it seems that you are using the MNIST dataset to evaluate, which has 10 characters (0-9), and not the ImageNet, which has 1001: 1000 classes plus a background. That would cause the error you're seeing. Try using the imagenet dataset and it should work!

**Comment by scotshinderman** [comment]
Ah right ... I was thinking the eval step would notice the datasets class size regardless of the training.   Thanks!
